packages/headless/src/features/facets/facet-set/facet-set-slice.ts

```diff
@@ -79,17 +79,17 @@ export const facetSetReducer = createReducer(
           return;
         }
 
-        const targetValue = facetRequest.currentValues.find(
+        const existingValue = facetRequest.currentValues.find(
           (req) => req.value === selection.value
         );
 
-        if (!targetValue) {
-          return;
+        if (existingValue) {
+          const isSelected = existingValue.state === 'selected';
+          existingValue.state = isSelected ? 'idle' : 'selected';
+        } else {
+          facetRequest.currentValues.push(selection);
```

Comment:
Two things if slightly unsure of:
- Should we care about the order the value is inserted? Or should we let the API take care of it, by removing freezeCurrentValues
- Now it takes a full value with selection, should we toggle the state on a new value as well as an existing one?

Reply:
> Should we care about the order the value is inserted? Or should we let the API take care of it, by removing freezeCurrentValues

I don't remember why `freezeCurrentValues` is necessary here. I would prefer to let the API choose the order.

> Now it takes a full value with selection, should we toggle the state on a new value as well as an existing one?

I feel like this issue stems from `toggleSelectFacetValue` being meant to be called with a value we read directly in the state, vs here we are expecting someone to create a value from scratch. I feel like the least ambiguous method would be to instead introduce a new action & function (e.g., `setFacetValueInitialState` or just `setFacetValueState`). I understand we took a decision to go with a `toggle` function, but we may be uncovering a better reason not to.

Reply:
First off, Benjamin's suggestion of a dedicated action could be a good option. If we take it, I would suggest `updateFacetCurrentValues` for the name.

Counter argument might be `toggleSelectCategoryFacetValue` creates new paths, so adjusting to create a missing value may make the behaviour more consistent with the category facet.


For creating a value inside toggleSelect, those are great questions.

1. We could treat it similar to selecting a facet search result (no control over position). There's logic in that reducer for inserting the value at the first idle index. It also adjusts the number of values if adding the new value takes the array over `numberOfValues`, otherwise the api will error. We should probably extract and reuse this code.

The freeze current values is tricky, but I think it _needs_ to be `false`, otherwise the facet will not "fill up" to the number of values? My reasoning being when `freeze` is `true`, it's for UX in response to a client interaction. However, an end-user cannot click on a non-existant value => if toggleSelect is adding, then it is most likely programmatic => no need to freeze?


2. When a toggleSelect value is being added, setting the initial state to what is given (i.e. not toggling) feels the most natural imo.

Reply:
Okay I think I'll go for being close to the behaviour of toggleSelectCategoryFacetValue & the facet search selection

When it doesn't exist
1. insert before the first idle value
2. don't touch freezeCurrentValues
3. don't toggle the state of the new value, take it as it is